Firestore/core/src/firebase/firestore/util/async_queue.cc

```diff
@@ -49,6 +50,10 @@ void AsyncQueue::VerifyIsCurrentQueue() const {
 }
 
 void AsyncQueue::ExecuteBlocking(const Operation& operation) {
+  // This is not guarded by `shut_down_mutex_` and `is_shutting_down_`
```

Comment:
Nit: I'd rather not mention `shut_down_mutex_` here, it's more of an implementation detail of how thread-safety is achieved.

Reply:
Done.